cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,319 @@
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	_ = cmd.MarkFlagRequired("for")
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Limit, err = listRequestLimit()
+	if err != nil {
+		return err
+	}
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	// parse --for flag
+	if o.UserSpecifiedForObject != "" {
+		mapper, err := clientGetter.ToRESTMapper()
+		if err != nil {
+			return err
+		}
+		var found bool
+		o.ForGVK, o.ForName, found, err = decodeResourceTypeName(mapper, o.UserSpecifiedForObject)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return fmt.Errorf("invalid value '%s' used in --for flag; value must be in the format [TYPE[.API-GROUP]/]NAME", o.UserSpecifiedForObject)
+		}
+
+		r := clientGetter.NewResourceBuilder().
+			Unstructured().
+			NamespaceParam(o.Namespace).
+			DefaultNamespace().
+			AllNamespaces(o.AllNamespaces).
+			FieldSelectorParam(fmt.Sprintf("metadata.name=%s", o.ForName)).
+			ResourceTypeOrNameArgs(true, o.ForGVK.Kind).
+			ContinueOnError().
+			Latest().
+			Flatten().
+			Do()
+		if r == nil {
+			return fmt.Errorf("Error building client for: %s/%s", o.ForGVK.Kind, o.ForName)
+		}
+		if err := r.Err(); err != nil {
+			return err
+		}
+		infos, err := r.Infos()
+		if err != nil {
+			return err
+		}
+
+		if len(infos) == 0 {
+			if !o.AllNamespaces {
+				fmt.Fprintf(o.ErrOut, "No resources found in %s namespace.\n", o.Namespace)
+			} else {
+				fmt.Fprintln(o.ErrOut, "No resources found")
+			}
+			return nil
+		}
+
+		job, ok := infos[0].Object.(*unstructured.Unstructured)
+		if !ok {
+			fmt.Fprintf(o.ErrOut, "Invalid object %+v. Unexpected type %T", job, infos[0].Object)
+			return nil
+		}
+		o.ForObject = job
+	}
+
+	return nil
+}
+
+func (o *PodOptions) ToPrinter(headers bool) (printers.ResourcePrinterFunc, error) {
+	if !o.PrintFlags.OutputFlagSpecified() {
+		printer := newPodTablePrinter().
+			WithNamespace(o.AllNamespaces).
+			WithNoHeaders(headers)
+		return printer.PrintObj, nil
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return nil, err
+	}
+
+	return printer.PrintObj, nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	var totalCount int
+
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	podLabelSelector, err := o.getPodLabelSelector()
+	if err != nil {
+		return err
+	}
```

Comment:
If users submit custom jobs using kueue, the list command never works fine.
So, could we add new arguments so that users can specify the arbitrary pod selectors for the custom job pod?

But, I'm ok with treating it as another feature. 

@alculquicondor WDYT?

Reply:
I created an issue for this #2806.

Reply:
@tenzen-y The implement of `getPodLabelSelector` return `"", nil` when the user sumbit their crd job type. And the label selector is support already.
The user case is that they have crd job type and use kueue to summit it, and you want to support to list their pod?
I'm a little confused.

Reply:
AFAIK, after my comment, this PR has been significantly refactored. So, I guess that my comment already is not valid.